walkingpadfitbit/interfaceadapters/walkingpad/treadmillcontroller.py

```diff
@@ -58,6 +59,19 @@ async def stop(self) -> None:
         await asyncio.sleep(3)
         await self.ctler.switch_mode(WalkingPad.MODE_STANDBY)
 
+    async def change_speed_by(
+        self,
+        speed_delta_kph: Annotated[float, Le(1.0), Ge(-1.0)],
+    ) -> float:
+        speed_before_kph: float = (
+            self.ctler.last_status.speed / 10 if self.ctler.last_status else 0.0
+        )
+
+        new_speed_kph = speed_before_kph + speed_delta_kph
+
+        await self.ctler.change_speed(int(new_speed_kph * 10))
+        return new_speed_kph
```

Comment:
Check what happens when it's 0.5 and we change it by -1.

also see if the new speed is already ready for us to use as the return value, instead of using our calculated one

Reply:
It was raising an error.

Updated the code to prevent it from setting a negative value.